src/main/java/org/eclipse/tractusx/ssi/lib/jwt/SignedJwtVerifier.java

```diff
@@ -60,19 +60,21 @@ public class SignedJwtVerifier {
    *
    * @param jwt a {@link SignedJWT} that was sent by the claiming party.
    * @return true if verified, false otherwise
-   * @throws JwtException the jwt exception
-   * @throws DidDocumentResolverNotRegisteredException the did document resolver not registered
-   *     exception
+   * @throws DidParseException
+   * @throws SignatureException
+   * @throws DidResolverException
+   * @throws SignatureVerificationFailedException
+   * @throws UnsupportedVerificationMethodException
    */
-  @SneakyThrows({JOSEException.class, DidResolverException.class})
   public boolean verify(SignedJWT jwt)
-      throws JwtException, DidDocumentResolverNotRegisteredException {
+      throws DidParseException, SignatureException, DidResolverException,
+          SignatureVerificationFailedException, UnsupportedVerificationMethodException {
 
     JWTClaimsSet jwtClaimsSet;
     try {
       jwtClaimsSet = jwt.getJWTClaimsSet();
     } catch (ParseException e) {
-      throw new JwtException(e);
+      throw new SignatureException(e.getMessage());
```

Comment:
should not throw java.security.SignatureException but rather SignatureParseException